### ISSUE TITLE
How to get both loss and model output at once, on a batch of data  ?

### ISSUE THREAD

**Issue by mohamed-ezz** [fchollet/keras#4397]
I'm using `train_on_batch` for training, which returns the loss on the given batch. How do I also get the output classification on that batch ? (I'd like to do some visualisations on the output)

To do that I currently do another call to `predict` to get the model output, but that's redundant since `train_on_batch` have already passed the input batch "forward".

Hopefully the solution would also work with `test_on_batch`.
